### PR TITLE
Start use rebar3_hank

### PR DESCRIPTION
**Pull request by vkatsuba**
* Added `rebar3_hank`
* Added few rules for ignore couple places
* Added `rebar3_hank`run into GH Actions
* Fixed `rebar3_hank` warnings

Total of warnings:
```erlang
===> The following pieces of code are dead and should be removed:
test/eradius_client_socket_test.erl:28: Field socket in record state is unused
test/eradius_client_socket_test.erl:28: Field client in record state is unused
test/eradius_client_SUITE.erl:114: Field subscribed_clients in record state is unused
test/eradius_client_SUITE.erl:113: Field sup in record state is unused
test/eradius_metrics_SUITE.erl:32: ?LOCALHOST is unused
test/eradius_logtest.erl:36: ?CLIENT_REQUESTS_COUNT is unused
test/eradius_logtest.erl:37: ?CLIENT_PROXY_REQUESTS_COUNT is unused
test/eradius_logtest.erl:39: ?NAS1_ACCESS_REQS is unused
test/eradius_logtest.erl:40: ?NAS2_ACCESS_REQS is unused
test/eradius_auth_SUITE.erl:60: ?CHALLENGE is unused
test/eradius_auth_SUITE.erl:77: ?SEND_SESSION_KEY40 is unused
test/eradius_auth_SUITE.erl:79: ?SEND_SESSION_KEY56 is unused
test/eradius_auth_SUITE.erl:81: ?SEND_SESSION_KEY128 is unused
src/eradius_server.erl:69: ?RESEND_TIMEOUT is unused
src/eradius_config.erl:15: ?valid_atom/1 is unused
src/eradius_config.erl:16: ?valid/1 is unused
applications/eradius_prometheus_collector/src/eradius_prometheus_collector.erl:12: ?METRIC_NAME_PREFIX is unused
src/eradius_server.erl:89: Callback validate_arguments/1 is not used anywhere in the module
test/eradius_proxy_SUITE.erl:148: strip_test/1 doesn't need its #1 argument
test/eradius_proxy_SUITE.erl:137: get_key_test/1 doesn't need its #1 argument
test/eradius_proxy_SUITE.erl:129: new_request_test/1 doesn't need its #1 argument
test/eradius_proxy_SUITE.erl:115: validate_options_test/1 doesn't need its #1 argument
test/eradius_proxy_SUITE.erl:68: validate_arguments_test/1 doesn't need its #1 argument
test/eradius_proxy_SUITE.erl:36: resolve_routes_test/1 doesn't need its #1 argument
test/eradius_metrics_SUITE.erl:174: radius_request/3 doesn't need its #3 argument
test/eradius_metrics_SUITE.erl:95: check_single_request/4 doesn't need its #4 argument
test/eradius_metrics_SUITE.erl:95: check_single_request/4 doesn't need its #3 argument
test/eradius_metrics_SUITE.erl:91: error_requests/1 doesn't need its #1 argument
test/eradius_metrics_SUITE.erl:83: bad_requests/1 doesn't need its #1 argument
test/eradius_metrics_SUITE.erl:74: good_requests/1 doesn't need its #1 argument
test/eradius_metrics_SUITE.erl:68: end_per_suite/1 doesn't need its #1 argument
test/eradius_logtest.erl:92: validate_arguments/1 doesn't need its #1 argument
test/eradius_logtest.erl:80: radius_request/3 doesn't need its #3 argument
test/eradius_logtest.erl:80: radius_request/3 doesn't need its #2 argument
test/eradius_lib_SUITE.erl:163: vendor_attribute_id_conflict_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:159: dec_vendor_ipv4_t/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:155: dec_vendor_string_t/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:151: dec_vendor_integer_t/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:147: dec_simple_ipv4_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:143: dec_simple_string_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:139: dec_simple_integer_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:132: enc_vendor_octet_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:127: enc_vendor_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:123: enc_ascend_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:117: enc_salt_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:113: enc_scramble_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:109: enc_simple_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:106: ascend_dec_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:103: ascend_enc_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:100: scramble_dec_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:97: scramble_enc_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:94: salt_decrypt_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:91: salt_encrypt_test/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:76: ipv6prefix/1 doesn't need its #1 argument
test/eradius_lib_SUITE.erl:63: end_per_suite/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:223: generate_ip_list_test/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:177: log_test/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:128: config_with_ranges/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:121: config_nas_removing/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:77: config_2/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:37: config_1/1 doesn't need its #1 argument
test/eradius_config_SUITE.erl:33: end_per_suite/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:219: send_request_failover/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:214: reconf_ports_10/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:209: reconf_ports_30/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:204: reconf_address/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:192: wanna_send/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:177: send_request/1 doesn't need its #1 argument
test/eradius_client_SUITE.erl:47: end_per_suite/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:135: mppe_start_key128_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:132: mppe_start_key56_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:129: mppe_start_key40_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:126: mppe_sendstart_key128_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:123: mppe_sendstart_key56_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:120: mppe_sendstart_key40_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:117: mppe_master_key_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:114: v2_authenticator_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:111: nt_response_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:108: des_key_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:105: password_hash2_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:102: password_hashhash_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:99: unicode_password_hash_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:96: password_hash_test/1 doesn't need its #1 argument
test/eradius_auth_SUITE.erl:92: unicode_test/1 doesn't need its #1 argument
include/eradius_lib.hrl:13: ?RAccess_Request is used only at src/eradius_lib.erl
include/eradius_lib.hrl:14: ?RAccess_Accept is used only at src/eradius_lib.erl
include/eradius_lib.hrl:15: ?RAccess_Reject is used only at src/eradius_lib.erl
include/eradius_lib.hrl:16: ?RAccounting_Request is used only at src/eradius_lib.erl
include/eradius_lib.hrl:17: ?RAccounting_Response is used only at src/eradius_lib.erl
include/eradius_lib.hrl:18: ?RAccess_Challenge is used only at src/eradius_lib.erl
include/eradius_lib.hrl:19: ?RDisconnect_Request is used only at src/eradius_lib.erl
include/eradius_lib.hrl:20: ?RDisconnect_Ack is used only at src/eradius_lib.erl
include/eradius_lib.hrl:21: ?RDisconnect_Nak is used only at src/eradius_lib.erl
include/eradius_lib.hrl:22: ?RCoa_Request is used only at src/eradius_lib.erl
include/eradius_lib.hrl:23: ?RCoa_Ack is used only at src/eradius_lib.erl
include/eradius_lib.hrl:24: ?RCoa_Nak is used only at src/eradius_lib.erl
include/eradius_lib.hrl:27: ?RUser_Name is used only at test/eradius_lib_SUITE.erl
include/eradius_lib.hrl:28: ?RUser_Passwd is used only at test/eradius_lib_SUITE.erl
include/eradius_lib.hrl:40: ?REAP_Message is used only at src/eradius_lib.erl
include/eradius_lib.hrl:41: ?RMessage_Authenticator is used only at src/eradius_lib.erl
```

### CODE REVIEW

File: include/eradius_lib.hrl [hunk 1]
```diff
@@ -9,23 +9,7 @@
 -type atom_address() :: {atom_name(), {atom_ip(), atom_port()}}.
 -type atom_address_pair() :: {atom_address(), atom_address()}.
 
-%%- cmds
--define(RAccess_Request,       1).
```
Comment (RoadRunnr) [inline include/eradius_lib.hrl:13]: No, eradius_lib.hrl is part of the library API, removing definitions from it is a NO-GO.

File: test/eradius_auth_SUITE.erl [hunk 2]
```diff
@@ -74,11 +71,8 @@ all() -> [
 %% MPPE keys
 -define(MASTER_KEY, <<16#FD, 16#EC, 16#E3, 16#71, 16#7A, 16#8C, 16#83, 16#8C, 16#B3, 16#88, 16#E5, 16#27, 16#AE, 16#3C, 16#DD, 16#31>>).
 -define(SEND_START_KEY40, <<16#8B, 16#7C, 16#DC, 16#14, 16#9B, 16#99, 16#3A, 16#1B>>).
--define(SEND_SESSION_KEY40, <<16#D1, 16#26, 16#9E, 16#C4, 16#9F, 16#A6, 16#2E, 16#3E>>).
 -define(SEND_START_KEY56, <<16#8B, 16#7C, 16#DC, 16#14, 16#9B, 16#99, 16#3A, 16#1B>>).
--define(SEND_SESSION_KEY56, <<16#D1, 16#5C, 16#00, 16#C4, 16#9F, 16#A6, 16#2E, 16#3E>>).
 -define(SEND_START_KEY128, <<16#8B, 16#7C, 16#DC, 16#14, 16#9B, 16#99, 16#3A, 16#1B, 16#A1, 16#18, 16#CB, 16#15, 16#3F, 16#56, 16#DC, 16#CB>>).
--define(SEND_SESSION_KEY128, <<16#40, 16#5C, 16#B2, 16#24, 16#7A, 16#79, 16#56, 16#E6, 16#E2, 16#11, 16#00, 16#7A, 16#E2, 16#7B, 16#22, 16#D4>>).
```
Comment (RoadRunnr) [inline test/eradius_auth_SUITE.erl:81]: rather the removing the defines, test cases that actually use those key should be added

File: test/eradius_metrics_SUITE.erl [hunk 3]
```diff
@@ -29,17 +29,17 @@
...
 
 init_per_suite(Config) ->
     application:load(eradius),
     EradiusConfig = [{radius_callback, ?MODULE},
-                     {servers, [{good,  {eradius_test_handler:localhost(ip), [1812]}},  %% for 'positive' responses, e.g. access accepts
-                                {bad,  {eradius_test_handler:localhost(ip), [1813]}},   %% for 'negative' responses, e.g. coa naks
-                                {error,  {eradius_test_handler:localhost(ip), [1814]}}  %% here things go wrong, e.g. duplicate requests
+                     {servers, [{good,  {?LOCALHOST, [1812]}},  %% for 'positive' responses, e.g. access accepts
+                                {bad,  {?LOCALHOST, [1813]}},   %% for 'negative' responses, e.g. coa naks
+                                {error,  {?LOCALHOST, [1814]}}  %% here things go wrong, e.g. duplicate requests
```
Comment (RoadRunnr) [inline test/eradius_metrics_SUITE.erl:42]: why? Not everything has to be hidden behind macros.

File: test/eradius_client_SUITE.erl [hunk 4]
```diff
@@ -151,8 +151,8 @@ test(false, Msg) ->
 
 check(OldState, NewState = #state{no_ports = P}, null, A) -> check(OldState, NewState, P, A);
 check(OldState, NewState = #state{socket_ip = A}, P, null) -> check(OldState, NewState, P, A);
-check(#state{sockets = OS, no_ports = _OP, idcounters = _OC, socket_ip = OA},
-        #state{sockets = NS, no_ports = NP, idcounters = NC, socket_ip = NA},
+check(#state{sockets = OS, no_ports = _OP, idcounters = _OC, socket_ip = OA, sup = _, subscribed_clients = _},
+        #state{sockets = NS, no_ports = NP, idcounters = NC, socket_ip = NA, sup = _, subscribed_clients = _},
```
Comment (RoadRunnr) [inline test/eradius_client_SUITE.erl:155]: why?

**Review by RoadRunnr — CHANGES_REQUESTED** [review]

**Comment by mgumz** [comment]
@vkatsuba in general: "add rebar3_hank" does not instantly ring a bell in my head on what all of this is good for.

(and, yes, after looking up i can deduce that rebar3_hank's purpose is to identify dead-code). but still: it's better to have another sentence like that in the PR description to set the context.

**Comment by vkatsuba** [comment]
> @vkatsuba in general: "add rebar3_hank" does not instantly ring a bell in my head on what all of this is good for.
> 
> (and, yes, after looking up i can deduce that rebar3_hank's purpose is to identify dead-code). but still: it's better to have another sentence like that in the PR description to set the context.

This PR is experiment and was provided for deep discussion with the team.

File: src/eradius_server.erl [hunk 5]
```diff
@@ -66,7 +66,6 @@
 -include("dictionary.hrl").
 -include("eradius_dict.hrl").
 
--define(RESEND_TIMEOUT, 5000).          % how long the binary response is kept after sending it on the socket
```
Comment (0xAX) [inline src/eradius_server.erl:69]: that should be used, not deleted

**Comment by vkatsuba** [comment]
Currently the `rebar3_hank` is provided a lot of noise. For this points was created few tickets and start discussion with `rebar3_hank` developers. For now current PR will be closed and will back to integration after fixing few issues in next release of `rebar3_hank` plugin.

**Pull request closed without merging**